### PR TITLE
chore: Add tests for streamx

### LINKED ISSUE #55: Add tests for streamx
Since we want to support streamx as the underlying gulp streams, we should add it to our tests.

### PR DESCRIPTION
**Pull request by phated**
Closes #55 

Blocked on a bug in one of these tests because streamx is causing a "premature close" error in `end-of-stream`.

cc @mafintosh

### CODE REVIEW

**Comment by sttk** [comment]
@phated This error might be due to `streamx`.

This error is caused at [this position](https://github.com/mafintosh/end-of-stream/blob/e104395e50015a6436d9747b4b1c2a617b267769/index.js#L54) in `end-of-stream`.

And the following code (using `streamx`) outputs that `stream._writableState.destroyed` is `undefined` though the stream was destroyed.
```
const streamx = require('streamx');
const stream = streamx.pipeline(
  streamx.Readable.from('foo bar baz'),
  new streamx.Transform({
    transform: function(data, cb) {
      cb(new Error('fail'));
    }
  }),
  new streamx.Writable(),
);
stream.on('error', function(err)  {});
stream.on('close', function(err) {
  console.log('[close]', 'stream.destroyed', stream.destroyed);
  console.log('[close]', 'stream._writableState.destroyed', stream._writableState.destroyed);
});
----
% node test.js    
[close] stream.destroyed true
[close] stream._writableState.destroyed undefined
```

On the other hand, the following code (using `pumpify`) outputs that `stream._writableState.destroyed` is `true`.
```
const pumpify = require('pumpify');
const stream = pumpify(
  streamx.Readable.from('foo bar baz'),
  new streamx.Transform({
    transform: function(data, cb) {
      cb(new Error('fail'));
    }
  }),
  new streamx.Writable(),
);
stream.on('error', function(err)  {});
stream.on('close', function(err) {
  console.log('[close]', 'stream.destroyed', stream.destroyed);
  console.log('[close]', 'stream._writableState.destroyed', stream._writableState.destroyed);
});
----
% node test.js 
[close] stream.destroyed true
[close] stream._writableState.destroyed true
```

Otherwise, `end-of-stream` should probably look at `stream.destroyed` instead of `stream._readableState.destroyed` and `stream._writableState.destroyed`.

**Comment by mafintosh** [comment]
I poked around your PR to see if this was a streamx issue, but it doesn't seem to be.

Internally you are calling the following in the failing test

```
eos(stream, { error: false }, callback)
```

That explicitly disables the error listener in streamx. It therefore doesn't get the "Fail" error, but streamx also doesn't end with the "everything was okay events" ie end/finish, so eos reports an premature close as the it only sees the close event.

This is by design and a good thing. If I missed something let me know.

**Comment by sttk** [comment]
@mafintosh Certainly, after removing `eosConfig = { error: false }` to `eos`, the test case which occured "premature close error" is passed.

@phated Can we remove this `eosConfig`? Even if it is done, all test cases are passed.

File: test/streams.js [hunk 1]
```diff
@@ -41,10 +41,6 @@ function pumpifyError() {
   var read = fs.createReadStream(exists);
   var pipeline = pumpify(through(), through(withErr), through());
 
-  pipeline.on('error', function (err) {
```
Comment (phated) [inline test/streams.js:44]: This looks like it was a workaround for the end-of-stream problem for normal streams.

**Comment by phated** [comment]
@mafintosh thanks for looking into that! I don't exactly remember why I wanted/needed the `{ error: false }` config, but all the tests seem to pass if I remove it (and I got to cleanup a workaround)

**Pull request merged**